### PR TITLE
Quick Start: Checklist improvements

### PR DESCRIPTION
**Pull request by nheagy**
Refs #9447

This adds several refinements to the Quick Start checklist that make it feature-complete.

checklist improvements

- Progress count added on blog details screen
- checkmarks added for completed items
- skip all button added (with alert to verify first)
- congratulations header added when entire list has been completed
- also make the suggestion timeout 10s because @SylvesterWilmott told me to

To test:
- Make a new site and turn one Quick Start, or ask me for my snippet to turn it on
- checkout each of the above items
- kinda make sure nothing else in quick start was broken

@ScoutHarris @aerych can one of you check this out tomorrow so it makes the code freeze?

### CODE REVIEW

**Comment by SylvesterWilmott** [comment]
Looks good @nheagy . What are the dimensions of the "Skip all" button?

**Comment by aerych** [comment]
Hey Nate!  A few notes:

- If I tap Skip All should I see the congratulations cell?  It seems weird to be congratulated for finishing when I skipped things.
- When I finish the last step I'm unable to return to the list.  Is there some other way I'd see the congratulations cell?  On android the list can be returned to one last time. 
- Once everything is completed (or skipped) we should probably hide the Skip All cell.  Right now its still present after I tap it.  I can scroll down the list and tap it multiple times if I want.

File: WordPress/Classes/ViewRelated/Blog/QuickStartCongratulationsCell.swift [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+    @IBOutlet var bottomLabel: UILabel?
+
+    override func awakeFromNib() {
+        super.awakeFromNib()
+
+        topLabel?.font = WPStyleGuide.fontForTextStyle(.subheadline, fontWeight: .semibold)
+        bottomLabel?.font = WPStyleGuide.fontForTextStyle(.subheadline, fontWeight: .regular)
+
+        topLabel?.text = NSLocalizedString("Congrats on finishing Quick Start  🎉", comment: "Headline shown to users when they complete all Quick Start items")
+        bottomLabel?.text = NSLocalizedString("doesn’t it feel good to cross things off a list?", comment: "subhead shown to users when they complete all Quick Start items")
```
Comment (aerych) [inline WordPress/Classes/ViewRelated/Blog/QuickStartCongratulationsCell.swift]: Should "doesn't" be capitalized?

**Comment by ScoutHarris** [comment]
Hey @nheagy . Not sure if this is related to this PR or not, but on iPad, 'View Site' cannot be tapped - it seems to be covered/blocked by the popup.

view_site_ipad

**Comment by ScoutHarris** [comment]
A couple more oddities on iPad:

1) If 'Skip All' is selected, the count is 6/7.
skip_all

2) After having selected 'Skip All', still on the Quick Start view, go to the site list and back to the same site. The Quick Start view is still visible, but the option is not in the menu.
quick_start_view

File: WordPress/Classes/ViewRelated/Blog/QuickStartCongratulationsCell.swift (hunk 1 above)
Comment (nheagy) [reply WordPress/Classes/ViewRelated/Blog/QuickStartCongratulationsCell.swift]: That's how both the mockup and WPAndroid have it, so I'll have to leave it

**Comment by nheagy** [comment]
Thanks for the reviews, @aerych & @ScoutHarris 

> If I tap Skip All should I see the congratulations cell?

Apparently—that's what's happening on Android.

> When I finish the last step I'm unable to return to the list. Is there some other way I'd see the congratulations cell? On android the list can be returned to one last time.

I'll fix that!

> Once everything is completed (or skipped) we should probably hide the Skip All cell.

Yes, I agree!

> on iPad, 'View Site' cannot be tapped

Well damn. That must have been introduced in a different PR so I'll address it in a new issue 😀 

> If 'Skip All' is selected, the count is 6/7.

If you can believe me, that's fixed in #10288 because it was breaking the detection for completion as well

> After having selected 'Skip All' [on iPad]

😬 All I'll say is Apple is lucky I really like iPads 😜

**Comment by nheagy** [comment]
@SylvesterWilmott the skip all button is the full table cell, so it's huge 😉

**Comment by nheagy** [comment]
Ok @aerych or @ScoutHarris this is ready for another look.

Skip all button now hides and the checklist can be viewed one last time.

**Comment by ScoutHarris** [comment]
Well I don't see anything else to complain about. :shipit: for me.

**Pull request merged**